src/wire.rs

```diff
@@ -259,6 +266,7 @@ impl<S: NetSession> NetTransport<S> {
             "read on terminated transport"
         );
         self.needs_flush = false;
+        // cloudhead: How do we know that the connection is established?
```

Comment:
Because of `reactor::Handler::handle_transport_event` with `SessionEvent::Established` sent to a service

Reply:
Not the session, I mean the tcp connection.